src/libraries/Microsoft.PowerFx.Core/IR/CoercionMatrix.cs

```diff
@@ -351,5 +356,30 @@ private static CoercionKind GetToStringCoercion(DType fromType)
                 return CoercionKind.None; // Implicit coercion?
             }
         }
+
+        private static CoercionKind GetUntypedObjectCoercion(DType toType)
+        {
+            switch (toType.Kind)
+            {
+                case DKind.String:
+                    return CoercionKind.UntypedToText;
+                case DKind.Boolean:
+                    return CoercionKind.UntypedToBoolean;
+                case DKind.Number:
+                    return CoercionKind.UntypedToNumber;
+                case DKind.Date:
+                    return CoercionKind.UntypedToDate;
+                case DKind.Time:
+                    return CoercionKind.UntypedToTime;
+                case DKind.DateTime:
+                    return CoercionKind.UntypedToDateTime;
+                case DKind.Color:
+                    return CoercionKind.UntypedToColor;
+                case DKind.Guid:
+                    return CoercionKind.UntypedToGUID;
+                default:
+                    return CoercionKind.None;
```

Comment:
>return CoercionKind.None; [](http://example.com/codeflow?start=20&length=25)

Should it be InvalidCoercionException, like for other invalid ones?

Reply:
The other similar functions in CoercionMatrix also return CoercionKind.None, it is not an exception, it just means that no coercion is generated and thus normal type checking will presumably generate an error.